lib/Serialization/Serialization.cpp

```diff
@@ -3333,44 +3351,97 @@ class Serializer::DeclSerializer : public DeclVisitor<DeclSerializer> {
       return false;
     }
 
-    return isDeserializationSafe(cast<ValueDecl>(decl));
-  }
-
-  static bool isDeserializationSafe(const ValueDecl *value) {
-    // A decl is safe if formally accessible publicly.
-    auto accessScope =
-        value->getFormalAccessScope(/*useDC=*/nullptr,
-                                    /*treatUsableFromInlineAsPublic=*/true);
-    if (accessScope.isPublic() || accessScope.isPackage())
-      return true;
-
-    if (auto accessor = dyn_cast<AccessorDecl>(value))
-      // Accessors are as safe as their storage.
-      if (isDeserializationSafe(accessor->getStorage()))
-        return true;
-
-    // Frozen fields are always safe.
-    if (auto var = dyn_cast<VarDecl>(value)) {
+    bool visitVarDecl(const VarDecl *var) {
       if (var->isLayoutExposedToClients())
         return true;
 
-      // Consider all lazy var storage as "safe".
+      // Consider all lazy var storage as externally accessible.
       // FIXME: We should keep track of what lazy var is associated to the
-      //        storage for them to preserve the same safeness.
+      //        storage for them to preserve the same accessibility.
       if (var->isLazyStorageProperty())
         return true;
```

Comment:
As the fixme says, this is something that likely marks too many things as safe/accessible. We should keep an eye on it.